From 483a36f61286ed2d351c23d44062fe52ab8832e4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 23 Aug 2017 23:37:42 +0200
Subject: [PATCH 1/1] efi_loader: decrement call level in efi_exit()

In efi_exit we leave the EFI service without calling EFI_EXIT.
We still have to decrement the call level.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_boottime.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 9cae3e3b0e..a60dbe0030 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -852,6 +852,9 @@ static efi_status_t EFIAPI efi_exit(efi_handle_t image_handle,
 	EFI_ENTRY("%p, %ld, %ld, %p", image_handle, exit_status,
 		  exit_data_size, exit_data);
 
+	/* Decrement call level */
+	__efi_exit_check();
+
 	loaded_image_info->exit_status = exit_status;
 	longjmp(&loaded_image_info->exit_jmp, 1);
 
-- 
2.14.1

